pkg/addons/device_plugin.go

```diff
@@ -192,3 +193,40 @@ func (n *NvidiaDevicePlugin) Manifest() []byte {
 func (n *NvidiaDevicePlugin) Deploy() error {
 	return applyDevicePlugin(n)
 }
+
+// A EFADevicePlugin deploys the EFA Device Plugin to a cluster
+type EFADevicePlugin struct {
+	rawClient kubernetes.RawClientInterface
+	region    string
+	planMode  bool
+}
+
+func (n *EFADevicePlugin) RawClient() kubernetes.RawClientInterface {
+	return n.rawClient
+}
+
+func (n *EFADevicePlugin) PlanMode() bool {
+	return n.planMode
+}
+
+func (n *EFADevicePlugin) Manifest() []byte {
+	return assetutil.MustLoad(efaDevicePluginYamlBytes)
+}
+
+func (n *EFADevicePlugin) SetImage(t *v1.PodTemplateSpec) error {
+	return useRegionalImage(t, n.region, efaResourceAccount)
+}
+
+// NewEFADevicePlugin creates a new EFADevicePlugin
+func NewEFADevicePlugin(rawClient kubernetes.RawClientInterface, region string, planMode bool) DevicePlugin {
+	return &EFADevicePlugin{
+		rawClient,
+		region,
+		planMode,
+	}
+}
+
+// Deploy deploys the EFA device plugin to the specified cluster
+func (n *EFADevicePlugin) Deploy() error {
+	return applyDevicePlugin(n)
```

Comment:
would be good to get some tests around this `applyDevicePlugin` func at some point.